odxtools/structures.py

```diff
@@ -121,16 +122,15 @@ def print_free_parameters_info(self) -> None:
         print(parameter_info(self.free_parameters), end="")
 
     def convert_physical_to_internal(self,
-                                     param_values: dict,
-                                     triggering_coded_request,
-                                     is_end_of_pdu=True):
-        logger.debug(f"{self.short_name} encode RPC"
-                     f" with params={param_values}")
-
+                                     param_values: ParameterValueDict,
+                                     triggering_coded_request: Optional[bytes],
+                                     is_end_of_pdu: bool = True) -> bytes:
         coded_rpc = bytearray()
         encode_state = EncodeState(
             coded_rpc,
             dict(param_values),
+            length_keys={},
+            table_keys={},
```

Comment:
Why  length_keys here?

Reply:
historical artifact: in a previous version of the patch I removed the mutable default values for `EncodeState` instead of making it a dataclass.